### PR TITLE
Renamed fields and types in result complexity aggregator

### PR DESCRIPTION
**Pull request by Sajan-99**
Hi,
This PR adds fields and type rename counts to the result complexity aggregator. 

Examples:

[ topLevelRenamedField: [ [id: 1], [id: 1] ]

Suppose the top level field name and type are renamed. Then there would be exactly 1 field rename, and 1 type rename.

[ topLevelRenamedField: [ [renamedId: 1], [renamedId: 1] ]

If the id field has also been renamed then there would be 3 field renames, and similarly for the type renames.

### CODE REVIEW

File: src/main/java/graphql/nadel/engine/ExecutionResultNodeMapper.java [hunk 1]
```diff
@@ -21,13 +24,16 @@
...
-    public ExecutionResultNode mapERNFromUnderlyingToOverall(ExecutionResultNode node, UnapplyEnvironment environment) {
+    public ExecutionResultNode mapERNFromUnderlyingToOverall(ExecutionResultNode node, UnapplyEnvironment environment, AtomicInteger typeRenameCount) {
 
         Map<String, String> typeRenameMappings = environment.typeRenameMappings;
         GraphQLSchema overallSchema = environment.overallSchema;
         ExecutionPath mappedPath = pathMapper.mapPath(node.getExecutionPath(), node.getResultKey(), environment);
         GraphQLObjectType mappedObjectType = mapObjectType(node, typeRenameMappings, overallSchema);
         GraphQLFieldDefinition mappedFieldDefinition = getFieldDef(overallSchema, mappedObjectType, node.getFieldName());
+
+        int typeDecrementValue = node instanceof ListExecutionResultNode ? - node.getChildren().size() : 0;
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/engine/ExecutionResultNodeMapper.java:35]: did you mean `negative node.getChildren().size()` here?

File: src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java [hunk 2]
```diff
@@ -99,22 +101,61 @@ private ExecutionResultNode convertImpl(ExecutionId executionId,
                                             NadelContext nadelContext,
                                             TransformationMetadata transformationMetadata) {
         final AtomicInteger nodeCount = new AtomicInteger();
-
-        HandleResult handleResult = convertSingleNode(root, null/*not for root*/, executionId, root, normalizedRootField, overallSchema, isHydrationTransformation, batched, fieldIdToTransformation, typeRenameMappings, onlyChildren, nadelContext, transformationMetadata, nodeCount);
+        final AtomicInteger fieldRenameCount = new AtomicInteger();
+        final AtomicInteger typeRenameCount = new AtomicInteger();
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java:105]: One thing to consider now that we have 3 counters is to invent a class to hold the mutative counters

eg

```
class ResultCounters {

     public addToFieldRenameCount(int amount)
     ...
```     

That way you have to only pass down 1 parameter to all the places (so the signatures look cleaner

And the logic of counters is encapsulated in one place

File: src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java [hunk 3]
```diff
@@ -130,18 +171,19 @@ private HandleResult convertRecursively(ExecutionResultNode node,
                                             boolean onlyChildren,
                                             NadelContext nadelContext,
                                             TransformationMetadata transformationMetadata,
-                                            AtomicInteger nodeCount) {
-        HandleResult handleResult = convertSingleNode(node, parentNode, executionId, root, normalizedRootField, overallSchema, isHydrationTransformation, batched, fieldIdToTransformation, typeRenameMappings, onlyChildren, nadelContext, transformationMetadata, nodeCount);
+                                            AtomicInteger nodeCount, AtomicInteger fieldRenameCount, AtomicInteger typeRenameCount) {
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java:174]: See here about an encapsulating class makes it nicer

File: src/main/java/graphql/nadel/engine/ExecutionResultNodeMapper.java (hunk 1 above)
Comment (Sajan-99) [reply src/main/java/graphql/nadel/engine/ExecutionResultNodeMapper.java:35]: yep this should be negative, i'll fix the spacing

**Review by faizan-siddiqui — CHANGES_REQUESTED** [review]
Yea I think after you have a ResultCounter class and pass that down and increment inside there, it looks good otherwise

File: src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java (hunk 3 above)
Comment (Sajan-99) [reply src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java]: done

File: src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java (hunk 2 above)
Comment (Sajan-99) [reply src/main/java/graphql/nadel/engine/ServiceResultNodesToOverallResult.java]: done

File: src/main/java/graphql/nadel/result/ResultCounter.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    public int getFieldRenameCount() {
+        return fieldRenameCount.get();
+    }
+
+    public int getTypeRenameCount() {
+        return typeRenameCount.get();
+    }
+
+}
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/result/ResultCounter.java:64]: Like it!

**Review by bbakerman — APPROVED** [review]
I just have one question on the negative increment in the list code

**Review by faizan-siddiqui — APPROVED** [review]

**Pull request merged**